### ISSUE TITLE
display inherited properties and methods in the outline

### ISSUE THREAD

**Issue by Osolemio44** [redhat-developer/vscode-java#2291]
I'm working with inheritance in classes and interfaces and I have to constantly go to the parents to see what properties and methods they have. Is there any way to see those elements inherited in the children in the outline?

**Comment by MidhunCN** [comment]
This would be a very nice feature to have.

**Comment by rgrunber** [comment]
I don't think we do this for the outline, mainly because the outline is implemented through [`textDocument/documentSymbol`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_documentSymbol), which uses a data type that assumes the symbols are all in the same document. There is no way to provide separate documents for the location of symbols. ~~That would have to be improved in the spec.~~ (**Update**: Forgot that this was discussed at https://github.com/microsoft/vscode-languageserver-node/pull/425 and seems they want to avoid such cases). We could just ignore the spec (with a setting), and try to do it anyways. Would probably have to override the outline view rather than relying on the LSP to populate it.

However, we did add the ability to view inherited symbols through the outline shown in the quick-pick menu. See https://github.com/redhat-developer/vscode-java/issues/2342 . From the "Go To Symbol In Editor" quick-pick, one just hits <kbd>ctrl</kbd>+<kbd>o</kbd> once again. The command is configurable (`java.action.showExtendedOutline`, "Java: Open Extended Outline")

https://github.com/user-attachments/assets/ebdc9128-d9bb-42fd-bc54-68b5fdb91480
